lib/spack/spack/database.py

```diff
@@ -748,7 +748,23 @@ def installed_extensions_for(self, extendee_spec):
         """
         for spec in self.query():
             try:
-                spack.store.layout.check_activated(extendee_spec, spec)
+                if spec.package.extends(extendee_spec):
+                    yield spec.package
+            except spack.directory_layout.NoSuchExtensionError:
+                continue
+            # TODO: conditional way to do this instead of catching exceptions
```

Comment:
why is catching exceptions a bad idea?

Reply:
This is copied from the old implementation. I imagine what is wanted is that instead of throwing an exception when a package doesn't extend a given spec is to return `False` or some other well-typed return value instead.

Reply:
Ah sorry, I mistook the code position. I actually have no idea where this comes into play. However in general I prefer python code that indicates violated assumptions by exceptions instead of using a special invalid marker of the normal return value. I'm not too sure why the exception is caught in the first place (and especially swallowed without a logged event..)

Reply:
Looking at the code that does this, it really should return a bool. Currently it throws or it doesn't.

Reply:
And this code doesn't need the catch, it is `check_activated` that does the weird control flow.